### PR TITLE
[GALL-3055] - Add 'Show more' button for the latest vrs grid on viewing rooms page.

### PR DESCRIPTION
**Pull request by oxaudo**
@pvinis and myself and later @damassi was graciously helping me to figure that one out. We got to the point that the page loads and all the queries are made properly, but unfortunately there seems to be no way to access data properly.

Some background here:
We have the `viewing-rooms` route that is defined in `ViewingRoom/routes.tsx` file that in the data spreads a fragment that is defined in `ViewingRoomsApp` which in turn uses fragment from `ViewingRoomsLatestGrid`. We were trying to add pagination to the `ViewingRoomsLatestGrid` data converting it to createPaginationContainer. So we made `ViewingRoomsLatestGrid` support `count` param and made sure that those params passed down from routes file.

However this does not really work. We can not access data in the child `ViewingRoomsLatestGrid` component. Pairing with @damassi he helped figure out that this will not work since Relay expects the property defined with the name on the route level and that property passed down to the child components. He said that we have a `viewer` pattern designed specifically for that in MP but that is also not obvious because VR data is stitched  from gravity and it's not clear how to add that data to the viewer.

@mzikherman - maybe we can get your brain on that to help us figure it out?

And just for illustration here is what happens now from this PR code:
Screen Shot 2020-07-16 at 2 35 16 PM

As we can see the `routes_ViewingRoomsAppQuery` is executed properly and returns `viewingRooms` in RelayResponce but those are not passed as prop to `ViewingRoomsApp` and then to `ViewingRoomsLatestGrid`.

### CODE REVIEW

File: src/v2/Apps/ViewingRoom/routes.tsx [hunk 1]
```diff
@@ -15,11 +15,14 @@ export const routes: RouteConfig[] = [
     prepare: () => {
       ViewingRoomsApp.preload()
     },
+    prepareVariables: () => {
```
[the remark this answers is not in the document]
Comment (pvinis) [reply src/v2/Apps/ViewingRoom/routes.tsx:18]: ohhhh we were so close!

**Comment by oxaudo** [comment]
Ok @pvinis - finally I got to work on that a bit more. 'Show more' button now works and I got the test to not crash. I guess we can add more specs for this show more functionality but I'm not exactly clear how  to set that up:(

File: src/v2/Apps/ViewingRoom/Components/ViewingRoomsLatestGrid.tsx [hunk 2]
```diff
@@ -116,36 +141,55 @@ export const ViewingRoomsLatestGrid: React.FC<ViewingRoomsLatestGridProps> = pro
             )
           })}
         </CSSGrid>
+        {hasMoreItems && (
+          <Flex flexDirection="column" alignItems="center">
+            <Button
+              variant="secondaryOutline"
+              size="medium"
+              onClick={() => loadMore()}
+              loading={isLoading ? true : false}
```
Comment (damassi) [inline src/v2/Apps/ViewingRoom/Components/ViewingRoomsLatestGrid.tsx:150]:
```
onClick={loadMore}
loading={isLoading}
```

File: src/v2/Apps/ViewingRoom/routes.tsx [hunk 3]
```diff
@@ -15,11 +15,17 @@ export const routes: RouteConfig[] = [
...
+      // Accomodates the grid of 3x items and 2x items well.
+      return { count: 12 }
+    },
     query: graphql`
-      query routes_ViewingRoomsAppQuery {
-        allViewingRooms: viewingRooms {
+      query routes_ViewingRoomsAppQuery($count: Int!, $after: String) {
+        allViewingRooms: viewer {
```
Comment (damassi) [inline src/v2/Apps/ViewingRoom/routes.tsx:24]: `viewer` strikes again 😄

**Comment by oxaudo** [comment]
@pvinis - ok to merge this you think?

**Comment by pvinis** [comment]
op. didnt see i was assigned. yea i think its good.

**Pull request merged**